app/Controllers/Http/ImageController.js

```diff
@@ -0,0 +1,46 @@
+"use strict";
+const fs = require("fs");
+const Env = use("Env");
+const AWS = require("aws-sdk");
+const credentials = new AWS.SharedIniFileCredentials({
+  profile: Env.get("AWS_PROFILE_NAME", "default"),
+});
+
+AWS.config.credentials = credentials;
+
+class ImageController {
+  // uploads image to S3, returns image url
+  async uploadImage({ request, response }) {
+    const { region } = request.all();
+
+    console.log("c")
+
+    AWS.config.update({ region });
+
+    const s3 = new AWS.S3();
+
+    try {
+      // get bucket name from environment variables
+      const bucket = Env.get("AWS_BUCKET_NAME");
+
+      const file = request.file("file");
+      // use file temp path to create a buffer
+      const buffer = await fs.promises.readFile(file.tmpPath);
+
+      const { Location: url } = await s3
+        .upload({
+          Key: file.clientName,
+          Bucket: bucket,
+          Body: buffer,
+          ACL: "public-read",
```

Comment:
I believe this requires an AWS CloudFront distribution, I implemented the code using the official AWS signer, though to test it, my account should be activated.

So here is two ways for now:
- I contacted AWS Support, I can wait for my account to be allowed to add CF distributions.
- Otherwise, I can push the commit here so you can test it manually, will add some debugging statements just in case it wont work.

Reply:
I can set up a CloudFront distro and S3 bucket and share the credentials with you offline.

Reply:
Yes, That would be appreciated.

Thank you.

Reply:
Update:

AWS Support solved the issue today, I created a CloudFront distribution and tested imageUpload feature, one modification was needed, now it is working fine.

NOTE: I used URLs signing with expire time, using RSA Public/Private key encryption ( see latest .env file ).